R/prevalentAbundance.R

```diff
@@ -0,0 +1,76 @@
+#' Prevalent abundance
+#'
+#' This function calculates the joint abundance of the prevalent taxa.
+#'
+#' @param x a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object
+#'
+#' @param abund_values A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to use for prevalence calculation.
+#'
+#' @inheritParams getPrevalentTaxa
+#'
+#' @details The core abundance index gives the relative proportion of the core
+#' species (in between 0 and 1). The core taxa are defined as those that exceed the
+#' given population prevalence threshold at the given detection level.
+#'
+#' @return
+#' a named \code{numeric} vector. It includes joint abundance of prevalent taxa,
+#' that is calculated for individual samples.
+#'
+#' @examples
+#' data(esophagus, package = "MicrobiomeExperiment")
+#' esophagus <- as(esophagus, "MicrobiomeExperiment")
+#' prevalentAbundance(esophagus)
+#'
+#' @name prevalentAbundance
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @export
+#'
+setGeneric("prevalentAbundance", signature = "x",
+           function(x, abund_values = "relabundance", ...)
+               standardGeneric("prevalentAbundance"))
+
+#' @rdname getPrevalence
+#' @export
+setMethod("prevalentAbundance", signature = c(x = "MicrobiomeExperiment"),
+          function(x, abund_values = "relabundance", detection = 0.1/100, prevalence = 50/100,
+                   include_lowest = FALSE, as_relative=FALSE, ...){
+
+              #Adds relative abundance table to the data
+              x <- relAbundanceCounts(x)
```

Comment:
No point in adding it, since you don't return it. Use the internal function `.calc_rel_abund` and add a dispatch for `ANY` on `getPrevalentTaxa`. See `getPrevalence` for an example